### ISSUE TITLE
Installing KDE Plasma 6 breaks existing GNOME installation

### ISSUE THREAD

**Issue by Pandapip1** [NixOS/nixpkgs#333747]
### Describe the bug

I had a GNOME installation (enabled with `services.xserver.desktopManager.gnome.enable = true`, `services.xserver.displayManager.gdm.enable = true`, and `services.xserver.displayManager.gdm.wayland = true`). I installed an additonal KDA Plasma 6 session (`services.desktopManager.plasma6.enable = true`, `programs.ssh.askPassword = lib.mkForce "${pkgs.seahorse}/libexec/seahorse/ssh-askpass"`) to test it out.

Doing so changed my GNOME window decorations and cursor, and reset my GNOME settings, even when in the GNOME session rather than the KDE Plasma session.

Obviously, as that wouldn't work for me, I disabled the Plasma session. But after rebooting, even more stuff broke. The window decorations didn't revert, and the cursor is literally a white box.

### Steps To Reproduce

See bug description.

### Expected behavior

No breakage

### Screenshots

TODO. Writing this on my laptop (unaffected).

### Additional context

*N.A.*

### Notify maintainers

@eclairevoyant, @K900, @SuperSandro2000, @jtojnar, @wegank (no module `meta.maintainers`, so notified last touchers).

### Metadata

TODO. Currently writing this on my laptop.

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
output here
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by eclairevoyant** [comment]
I don't know anything about plasma or gnome, you probably want to @ `NixOS/gnome` or `NixOS/qt-kde`

**Comment by Pandapip1** [comment]
I'm not sure whose fault it is, so I'll ping the @NixOS/gnome folks, since they'll probably know at least how to clean up this mess in the meantime.

**Comment by MatthewCash** [comment]
I don't think this is a NixOS issue. Both Plasma and Gnome use the same GTK config (`$XDG_CONFIG_HOME/gtk-*`), so when you update your Plasma theme it will modify the same GTK config that Gnome apps use (IIRC [kdePackages.kde-gtk-config](https://search.nixos.org/packages?channel=unstable&from=0&size=50&sort=relevance&type=packages&query=kde-gtk-config) does it).

You might be able to do some shell script trickery to copy a pre-made GTK config to that location depending on which desktop you choose at login to have separate themes for each desktop.

**Comment by Pandapip1** [comment]
> I don't think this is a NixOS issue

This is fair, but I feel like maybe there should be a warning if both are enabled at the same time?

Also, where should I open the feature requests in the GNOME & KDE repos? I assume for GNOME, it would be the [gnome-shell issue tracker](https://gitlab.gnome.org/GNOME/gnome-shell/-/issues)?

**Comment by MatthewCash** [comment]
> This is fair, but I feel like maybe there should be a warning if both are enabled at the same time.

Might be a good idea, but I don't see anything technically wrong with running both, you just can't rely on Plasma or Gnome to manager your GTK theme anymore.

> Also, where should I open the feature requests in the GNOME & KDE repos? I assume for GNOME, it would be the [gnome-shell issue tracker](https://gitlab.gnome.org/GNOME/gnome-shell/-/issues)?

I doubt either Gnome or KDE would classify this as a bug since both desktops are functioning as intended, and I'm not sure what the desktops themselves could even do to fix this. The issue is that GTK applications check `$XDG_CONFIG_HOME/gtk-*` for the GTK theme, but this file is per-user, not per-desktop so you are intended to have one GTK theme for all of your desktops. Anything beyond that is the wild west.

If you want this behavior to be "fixed", you are going to have to find a way to automatically change that config file depending on what desktop you login to.

**Comment by Pandapip1** [comment]
> Might be a good idea, but I don't see anything technically wrong with running both, you just can't rely on Plasma or Gnome to manage your GTK theme anymore.

I didn't expect it to completely nuke my existing one though...

**Comment by MatthewCash** [comment]
> I didn't expect it to completely nuke my existing one though...

Your existing Gnome install shouldn't be completely toast, I suspect you can just go into Gnome Tweaks and set your GTK theme to Adwaita again.

**Comment by Pandapip1** [comment]
> Your existing Gnome install shouldn't be completely toast, I suspect you can just go into Gnome Tweaks and set your GTK theme to Adwaita again.

Well I accidentally just `rm -rf ~`'d, so if it wasn't toast then, it is now. At least it solved the issues?

**Comment by jtojnar** [comment]
I would say this is KDE bug/misfeature. Apps should not be changing user’s config without their consent.

**Comment by dotlambda** [comment]
> > I didn't expect it to completely nuke my existing one though...
> 
> Your existing Gnome install shouldn't be completely toast, I suspect you can just go into Gnome Tweaks and set your GTK theme to Adwaita again.

Indeed, GNOME Tweaks has an option to reset everything to the default. That will restore the cursor.

**Comment by MatthewCash** [comment]
> I would say this is KDE bug/misfeature. Apps should not be changing user’s config without their consent.

I *think* the user has to change their theme in Plasma's system settings for this to happen, so I wouldn't say it's against the user's consent. It's just not immediately obvious that changing the GTK theme in Plasma affects all desktops. Changing your GTK theme in Gnome Tweaks will also affect GTK in Plasma, so it's not just a Plasma-specific quirk.

**Comment by Pandapip1** [comment]
> I think the user has to change their theme in Plasma's system settings for this to happen, so I wouldn't say it's against the user's consent.

Nope, this happened automatically when I started a plasma session through gdm, without me otherwise changing any settings.

**Comment by MatthewCash** [comment]
> Nope, this happened automatically when I started a plasma session through gdm, without me otherwise changing any settings.

Oh yeah that does make sense too. Plasma would want to set your GTK theme to Breeze immediately otherwise apps would look out of place until you manually change your settings.

I wonder if there is some type of `GTK_HOME` variable that could be set so that the Plasma session can have its own GTK config location.

**Comment by Pandapip1** [comment]
> I wonder if there is some type of GTK_HOME variable that could be set so that the Plasma session can have its own GTK config location.

There appears to be https://docs.gtk.org/gtk4/running.html#gtk_theme, but it seems that the icon theme spec doesn't allow for overriding where the icon theme is located.

**Comment by MatthewCash** [comment]
> There appears to be https://docs.gtk.org/gtk4/running.html#gtk_theme, but it seems that the icon theme spec doesn't allow for overriding where the icon theme is located.

I was hoping for one that could be used to override the entire GTK config dir, something like the old `GTK2_RC_FILES` variable, but I don't think one exists for GTK 3 :/
